### PR TITLE
Next Week On Leave Employees API

### PR DESCRIPTION
**Pull request by Anoyaa**

### CODE REVIEW

File: src/ui/src/app/Service/leave-management.service.ts [hunk 1]
```diff
@@ -29,14 +25,20 @@ export class LeaveManagementService {
...
+  id= this.userService.userId;
   private baseUrl = environment.leave;
+  
   getLeaveBalance(id: number) {
     return this.http.get<Array<ILeaveBalanceList>>(`${this.baseUrl}/leave-balance?EmployeeId=${id}`);
   }
 
-  getLeaveRequestHistory(id: number, pageNumber: number, pageSize: number): Observable<ILeaveRequestHistoryResponse> {
-    return this.http.get<ILeaveRequestHistoryResponse>(`${this.baseUrl}/leave-request?EmployeeId=${id}&PageNumber=${pageNumber}&PageSize=${pageSize}`); 
+  getLeaveRequestHistory(id: number, pageNumber: number, pageSize: number, selectedStatus: number) {
```
Comment (zameelnp) [inline src/ui/src/app/Service/leave-management.service.ts:35]: make selectedStatus as nullable

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/leave-history/leave-history.component.html [hunk 2]
```diff
@@ -25,32 +36,48 @@
       <td>{{leaveRequest.fromDate}}</td>
       <td>{{leaveRequest.toDate}}</td>
       <td>{{leaveRequest.halfDay}}</td>
-      <td>{{leaveRequest.description}}</td>
+      <td><a href="javascript:void()" (click)="viewDescription(leaveRequest.description)">View </a></td>
```
Comment (zameelnp) [inline src/ui/src/app/dashboard/dashboard-body/main/leave-management/leave-history/leave-history.component.html:39]: purpose?

Comment (Anoyaa) [reply src/ui/src/app/dashboard/dashboard-body/main/leave-management/leave-history/leave-history.component.html:39]: for the purpose of shortening the description to show description as a popup Modal when clicked

File: src/services/WolfDen.Application/Requests/Queries/LeaveManagement/LeaveRequests/GetLeaveRequestHistory/GetLeaveRequestHistoryQueryHandler.cs [hunk 3]
```diff
@@ -42,10 +37,27 @@ public async Task<LeaveRequestHistoryResponseDto> Handle(GetLeaveRequestHistoryQ
...
                 .ToListAsync(cancellationToken);
 
+
+            if (request.LeaveStatusId.HasValue)
+            {
+                filteredData = leaveRequestList.Where(x => x.LeaveRequestStatusId.Equals(request.LeaveStatusId)).ToList();
+            }
+            else
+            {
+                filteredData = leaveRequestList;
```
[the remark this answers is not in the document]
Comment (Anoyaa) [reply src/services/WolfDen.Application/Requests/Queries/LeaveManagement/LeaveRequests/GetLeaveRequestHistory/GetLeaveRequestHistoryQueryHandler.cs]: updated

File: src/services/WolfDen.Application/Requests/Queries/LeaveManagement/LeaveRequests/GetLeaveRequestHistory/GetLeaveRequestHistoryQueryHandler.cs [hunk 4]
```diff
@@ -42,10 +37,27 @@ public async Task<LeaveRequestHistoryResponseDto> Handle(GetLeaveRequestHistoryQ
                 })
                 .ToListAsync(cancellationToken);
 
+
+            if (request.LeaveStatusId.HasValue)
+            {
+                filteredData = leaveRequestList.Where(x => x.LeaveRequestStatusId.Equals(request.LeaveStatusId)).ToList();
```
[the remark this answers is not in the document]
Comment (Anoyaa) [reply src/services/WolfDen.Application/Requests/Queries/LeaveManagement/LeaveRequests/GetLeaveRequestHistory/GetLeaveRequestHistoryQueryHandler.cs]: updated

File: src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs [hunk 5]
```diff
@@ -24,6 +25,11 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
             return await _mediator.Send(query,cancellationToken); 
         }
 
+        [HttpGet("on-leave")]
```
[the remark this answers is not in the document]
Comment (Anoyaa) [reply src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs]: updated

File: src/ui/src/app/Service/leave-management.service.ts (hunk 1 above)
Comment (Anoyaa) [reply src/ui/src/app/Service/leave-management.service.ts]: ok,updated

**Pull request merged**